code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -280,8 +289,18 @@
 
 	if(mind)
 		mind.transfer_to(new_xeno)
+		if(new_caste_type == /mob/living/carbon/xenomorph/hivemind) //THERE ARE BETTER METHODS OF DOING THIS, THIS WAS JUST THE FASTEST AND LESS STRESSFUL WAY
+			var/mob/living/carbon/xenomorph/hivemind/newmind = new_xeno
+			var/obj/effect/alien/weeds/node/hivemindcore/newcore = new /obj/effect/alien/weeds/node/hivemindcore(new_xeno.loc)
+			newcore.parent = newmind
+			newmind.core = newcore
 	else
 		new_xeno.key = key
+		if(new_caste_type == /mob/living/carbon/xenomorph/hivemind)
+			var/mob/living/carbon/xenomorph/hivemind/newmind = new_xeno
+			var/obj/effect/alien/weeds/node/hivemindcore/newcore = new /obj/effect/alien/weeds/node/hivemindcore(new_xeno.loc)
+			newcore.parent = newmind
+			newmind.core = newcore
```

Comment:
This is duplicated code.
Add a `post_evolve()` redefined proc or something.

Reply:
Actually redefining it won't work.
The ideal way would be doing a xeno caste `on_creation()` proc or the likes, but if you don't feel like refactoring this just add this code after the `if-else`, since it's happening anyway on the same format.